### PR TITLE
Add LLMSpec class prototype

### PR DESCRIPTION
**Pull request by tddschn**
Add LLMSpec class prototype at https://github.com/tensorchord/llmspec/blob/dev/llmspec/LLMSpec.py

### CODE REVIEW

File: llmspec/LLMSpec.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import json
+
+from pydantic import BaseModel
+
+
+class LLMSpec(BaseModel):
+    """
+    # TODO: docstring
+    Example usage
+    llm = LLM(text="你好")
```
Comment (kemingy) [inline llmspec/LLMSpec.py]: Avoid non-ASCII char.

File: llmspec/LLMSpec.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import json
+
+from pydantic import BaseModel
+
+
+class LLMSpec(BaseModel):
```
Comment (kemingy) [inline llmspec/LLMSpec.py]: Never used `BaseModel`?

File: llmspec/__init__.py
Comment (kemingy) [inline llmspec/__init__.py:1]: Expose here.

File: pyproject.toml [hunk 3]
```diff
@@ -1,17 +1,34 @@
-[tool.pdm]
-
 [project]
 name = "llmspec"
 version = "0.1.0"
 description = "LLM unified interface"
 authors = [
-    {name = "Teddy Xinyuan Chen", email = "45612704+tddschn@users.noreply.github.com"},
+    { name = "Teddy Xinyuan Chen", email = "45612704+tddschn@users.noreply.github.com" },
```
Comment (kemingy) [inline pyproject.toml]: Use company info if the copyright belongs to the company.

File: pyproject.toml [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
 ]
 dependencies = []
-requires-python = ">=3.6"
+requires-python = ">=3.7"
 readme = "README.md"
-license = {text = "apache2"}
+license = { text = "apache2" }
```
Comment (kemingy) [inline pyproject.toml]: Use the standard SPDX name.

File: llmspec/LLMSpec.py
Comment (kemingy) [inline llmspec/LLMSpec.py:1]: Use the lowercase.

File: llmspec/LLMSpec.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    """
+    # TODO: docstring
+    Example usage
+    llm = LLM(text="你好")
+    encoded = llm.encode()
+    print(f"Encoded: {encoded}")
+
+    decoded = LLM.decode(encoded)
+    print(f"Decoded: {decoded}")
```
Comment (kemingy) [inline llmspec/LLMSpec.py]: Wrong format of code in the docstrings.

File: llmspec/LLMSpec.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    def encode(self) -> str:
+        data = {"text": self.text}
+        return json.dumps(data)
+
+    @staticmethod
+    def decode(data: str | bytes, encoding: str = "utf-8") -> "LLMSpec":
+        if isinstance(data, bytes):
+            data = data.decode(encoding)
+        decoded_data = json.loads(data)
+        return LLMSpec(**decoded_data)
```
Comment (kemingy) [inline llmspec/LLMSpec.py]: Use the normal way.

File: llmspec/LLMSpec.py (hunk 2 above)
Comment (tddschn) [reply llmspec/LLMSpec.py]: It helps validate the attribute types?

File: pyproject.toml [hunk 7]
```diff
@@ -1,17 +1,34 @@
-[tool.pdm]
-
 [project]
 name = "llmspec"
 version = "0.1.0"
 description = "LLM unified interface"
 authors = [
-    {name = "Teddy Xinyuan Chen", email = "45612704+tddschn@users.noreply.github.com"},
+    { name = "TensorChord", email = "45612704+tddschn@users.noreply.github.com" },
```
Comment (kemingy) [inline pyproject.toml]: Email

File: llmspec/LLMSpec.py (hunk 2 above)
Comment (kemingy) [reply llmspec/LLMSpec.py]: Not in the dependencies. There are some performance issues with v1*.

File: llmspec/LLMSpec.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    Example usage
+    llmspec = LLMSpec(text="Hello")
+    encoded = llmspec.encode()
+    print(f"Encoded: {encoded}")
+
+    decoded = llmspec.decode(encoded)
+    print(f"Decoded: {decoded}")
+
+    to_model = llmspec.to_model(name="moss")
+    print(f"MOSS format: {to_model}")
```
Comment (kemingy) [inline llmspec/LLMSpec.py]: Wrong format.

**Review by kemingy — CHANGES_REQUESTED** [review]

File: llmspec/LLMSpec.py
Comment (kemingy) [reply llmspec/LLMSpec.py:1]: filename

Comment (tddschn) [reply llmspec/LLMSpec.py:1]: I was on macOS (case insensitive FS) and renamed the file to lower case in VSCode's Explorer pane, which didn't work (but I didn't know).  
I'll rename using git commands.

Comment (tddschn) [reply llmspec/LLMSpec.py:1]: git mv didn't work either, I'll remove the file and re-recreate it.

**Review by kemingy — APPROVED** [review]

**Pull request merged**